superset-frontend/packages/superset-ui-core/src/color/utils.ts

```diff
@@ -51,3 +52,22 @@ export function getContrastingColor(color: string, thresholds = 186) {
 
   return r * 0.299 + g * 0.587 + b * 0.114 > thresholds ? '#000' : '#FFF';
 }
+
+export function getAnalogousColors(colors: string[], results: number) {
+  const generatedColors: string[] = [];
+  const ext = 3;
+  const analogousColors = colors.map(color => {
+    const result = tinycolor(color).analogous(results + ext);
+    return result.slice(ext);
+  });
+
+  // [[A, AA, AAA], [B, BB, BBB]] => [A, B, AA, BB, AAA, BBB]
```

Comment:
I appreciate the explanation of this interleaving. This is one of those little bits of code where there are 90 ways to do it. We might be able to squeeze some performance here, but I don't think anyone's palette will be big enough to cause a perceivable performance problem anway.